homeassistant/helpers/intent.py

```diff
@@ -539,12 +573,19 @@ async def async_handle_states(
     async def async_call_service(self, intent_obj: Intent, state: State) -> None:
         """Call service on entity."""
         hass = intent_obj.hass
+
+        extra_attrs: dict[str, Any] = {}
+        if self.extra_slot_names:
+            extra_attrs = {
+                key: intent_obj.slots[key]["value"] for key in self.extra_slot_names
+            }
+
         await self._run_then_background(
             hass.async_create_task(
                 hass.services.async_call(
                     self.domain,
                     self.service,
-                    {ATTR_ENTITY_ID: state.entity_id},
+                    {ATTR_ENTITY_ID: state.entity_id, **extra_attrs},
```

Comment:
Instead of making a copy, why not define `attr` before checking the extra attributes?

Reply:
Done.